### DISCUSSION TITLE
GitHub education I can’t upload picture

### DISCUSSION THREAD

**Discussion by Noel5027** [community/community#143918] [GitHub Education]
### Hi everyone,

When I upload a photo of a document, only the front camera works, and if I switch to the rear camera, I only see a white screen.
There is no file upload button. I tried everything instructed on the github education forum, but it didn't work.

**Comment by lllHunterlllSV** [comment]
**Hello! 👋**

Thank you for sharing the issue you're facing! It sounds like you're dealing with a frustrating technical problem, and I’m here to help you troubleshoot. Let’s go through some possible solutions:

- 📸 **Camera Permissions**: Make sure your device's camera permissions are correctly set up. Sometimes, the rear camera may not work due to restrictions or permissions issues:
    - **iPhone/iPad**: Go to *Settings* > *Privacy* > *Camera*, and ensure that your browser or GitHub Education app has permission to use the camera.
    - **Laptop/Desktop**: Check your browser settings to ensure the camera is allowed. You can usually check this by clicking on the camera icon in the browser's address bar when you're on the GitHub Education registration page.

- 🖼️ **File Upload Button Missing**: If you cannot see the file upload button, try the following:
    - **Clear Browser Cache**: Sometimes, cached data can cause issues with buttons and other page elements. Clear your browser's cache or try using **Incognito/Private Mode**.
    - **Browser Compatibility**: Make sure you're using an updated version of a supported browser (like **Chrome**, **Safari**, or **Firefox**). Sometimes, certain features might not display correctly on older browsers.
    - **Switch Devices**: If you're still having trouble, try using a different device, if possible. This can help determine if the issue is specific to one device.

- 🔄 **Switching to Rear Camera**: Since you mentioned that only the front camera works, it could be an issue with the camera interface in the GitHub Education registration process. If you cannot use the rear camera, you could try:
    - Ensure that you are holding the document close enough for the front camera to capture it clearly.
    - **Contact GitHub Support**: If none of the above solutions work, the next step would be to contact [GitHub Support](https://support.github.com/). They may be able to troubleshoot further or provide an alternative way to upload your document.

I hope one of these solutions works for you! Let me know if you need more assistance. Best of luck with your GitHub Education registration! 😊

**Reply by michlbro** [reply]
Hello, I am also having the same issue (IOS 18.0.1, iPhone 14 Plus, Safari). 

One thing I can comment on is that the 1MB photo limit is not possible in this era. Most people have a phone capable of 4K and for some reason Apple does not allow any resolution lower, hence it will always be larger than 1MB. 

The "camera in use" icon shows up for both front/rear and I have tried checking camera permissions, but not other troubleshoot steps. I feel like there's a bug in the code for the rear camera since "wide/short" also lenses came up in the dropdown, which is a first for me when trying to take a photo of any ID. Either way, if I were to use the rear camera it would have probably surpassed this 1MB limit. 

I ended up using my front camera using spirit leveller to take the photo (It did get approved so the hard work paid off) 😂
